statsmodels/robust/scale.py

```diff
@@ -81,6 +85,73 @@ def iqr(a, c=Gaussian.ppf(3/4) - Gaussian.ppf(1/4), axis=0):
         return np.squeeze(np.diff(quantiles, axis=0) / c)
 
 
+def qn_scale(a, c=1 / (np.sqrt(2) * Gaussian.ppf(5 / 8)), axis=0):
+    """
+    Computes the Qn robust estimator of scale, a more efficient alternative
+    to the MAD. The Qn estimator of the array a of length n is defined as
+    c * {abs(a[i] - a[j]): i<j}_(k), for k equal to [n/2] + 1 choose 2. Thus,
+    the Qn estimator is the k-th order statistic of the absolute differences
+    of the array. The optional constant is used to normalize the estimate
+    as explained below. The implementation follows the algorithm described
+    in Croux and Rousseeuw (1992).
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.
+    c : float, optional
+        The normalization constant. The default value is used to get consistent
+        estimates of the standard deviation at the normal distribution.
+    axis : int, optional
+        The default is 0.
+
+    Returns
+    -------
+    The Qn robust estimator of scale
+    """
+    a = array_like(a, 'a', ndim=None, dtype=np.float64, contiguous=True,
+                   order='C')
+    c = float_like(c, 'c')
+    if a.ndim == 0:
+        raise ValueError("a should have at least one dimension")
+    elif a.size == 0:
+        return np.nan
+    else:
+        return np.apply_along_axis(_qn, axis=axis, arr=a, c=c)
```

Comment:
Rather than returning firectly, set this to a variable, e.g., 

```
out = np.apply...
if out.ndim == 0:
    return float(out)
return out
```

So that the 1-d reduction returns a float.

Reply:
Done in b04b51cb2b83b3d9e1828d3b53956c1a7fd44629